verifiers/internal/gha/provenance_forgeable.go

```diff
@@ -9,13 +9,21 @@ import (
 	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/common"
 	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/iface"
 	slsav02 "github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/v0.2"
+	slsav1 "github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/v1.0"
 )
 
 func verifyProvenanceMatchesCertificate(prov iface.Provenance, workflow *WorkflowIdentity) error {
```

Comment:
This isn't new with this change, but it seems this function assumes that the provenance is going to be generated by GitHub but its possible that it comes from elsewhere (e.g. GitLab https://registry.npmjs.org/-/npm/v1/attestations/@ps-testing%2fgitlab-npm-provenance@1.0.19). Is my interpretation correct? Are there any plans to support verifying provenance that doesn't come from GitHub?

Reply:
OK that makes sense. My only other concern would be making sure the errors the library returns if you try to verify an unsupported type of provenance (e.g. gitlab v0.2) are clear. This switch statement defaults to calling `verifySystemParameters` for any type that isn't `slsav1.NpmCLIGithubActionsProvenance` (with `verifySystemParameters` assuming its a GitHub attestation). Another way to do it would be to check for all expected types and then default to returning an some "unsupported type" error, to make it clear to external clients what's going on. But maybe this is handled elsewhere in the code before it gets to this function?

Reply:
Yes, earlier in the code, the codepath is routed based on BuilderID, 
 - https://github.com/slsa-framework/slsa-verifier/blob/96619e48c27f7a0aa7eec2409ffa315656ec7e15/verifiers/verifier.go#L26-L32
 - https://github.com/slsa-framework/slsa-verifier/blob/96619e48c27f7a0aa7eec2409ffa315656ec7e15/verifiers/internal/gha/verifier.go#L42-L45

and will error for unsupported builders, like those from GitLab.
 - https://github.com/npm/cli/blob/22731831e22011e32fa0ca12178e242c2ee2b33d/workspaces/libnpmpublish/lib/provenance.js#L74C31-L74C45

Reply:
OK awesome thank you :)